packages/full-stack-tests/src/hierarchy-builder/grouping/BaseClassGrouping.test.ts

```diff
@@ -0,0 +1,764 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { PhysicalPartition, Subject } from "@itwin/core-backend";
+import { IModel } from "@itwin/core-common";
+import { IModelConnection } from "@itwin/core-frontend";
+import { SchemaContext } from "@itwin/ecschema-metadata";
+import { ECSchemaRpcLocater } from "@itwin/ecschema-rpcinterface-common";
+import { createECSqlQueryExecutor, createMetadataProvider } from "@itwin/presentation-core-interop";
+import { HierarchyNode, HierarchyProvider, IHierarchyLevelDefinitionsFactory, NodeSelectClauseFactory } from "@itwin/presentation-hierarchy-builder";
+import { buildIModel, insertPhysicalPartition, insertSubject } from "../../IModelUtils";
+import { initialize, terminate } from "../../IntegrationTests";
+import { NodeValidators, validateHierarchy } from "../HierarchyValidation";
+
+describe("Stateless hierarchy builder", () => {
+  describe("BaseClassGrouping", () => {
+    let selectClauseFactory: NodeSelectClauseFactory;
+    let subjectClassName: string;
+    let physicalPartitionClassName: string;
+
+    before(async function () {
+      await initialize();
+      subjectClassName = Subject.classFullName.replace(":", ".");
+      physicalPartitionClassName = PhysicalPartition.classFullName.replace(":", ".");
+      selectClauseFactory = new NodeSelectClauseFactory();
+    });
+
+    after(async () => {
+      await terminate();
+    });
+
+    function createProvider(props: { imodel: IModelConnection; hierarchy: IHierarchyLevelDefinitionsFactory }) {
+      const { imodel, hierarchy } = props;
+      const schemas = new SchemaContext();
+      schemas.addLocater(new ECSchemaRpcLocater(imodel.getRpcProps()));
+      const metadataProvider = createMetadataProvider(schemas);
+      return new HierarchyProvider({
+        metadataProvider,
+        hierarchyDefinition: hierarchy,
+        queryExecutor: createECSqlQueryExecutor(imodel),
+      });
+    }
```

Comment:
This is repeated across all test suites - let's move it somewhere so it can be shared.

Reply:
Moved it to `IModelUtils.ts` file